diff --git a/WORKSPACE b/WORKSPACE
index c58e76b6..259787c1 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -83,6 +83,21 @@ http_archive(
        "https://github.com/google/ruy/archive/9f53ba413e6fc879236dcaa3e008915973d67a4f.zip",
    ],
 )
+# Emscripten
+http_archive(
+   name = "emsdk",
+   strip_prefix = "emsdk-3.0.0/bazel",
+   #sha256 = "8439411fe684a3f59d387b84fc1699c494405d3b5001c48b19ad681fe441bcce",
+   urls =[
+        "https://github.com/emscripten-core/emsdk/archive/3.0.0.tar.gz",
+   ]
+)
+
+load("@emsdk//:deps.bzl", emsdk_deps = "deps")
+emsdk_deps()
+
+load("@emsdk//:emscripten_deps.bzl", emsdk_emscripten_deps = "emscripten_deps")
+emsdk_emscripten_deps(emscripten_version = "3.0.0")
 
 # Android NDK location and version is auto-detected from $ANDROID_NDK_HOME environment variable
 android_ndk_repository(name = "androidndk")
diff --git a/emscripten.bzl b/emscripten.bzl
index 130d5f16..c0fb9582 100644
--- a/emscripten.bzl
+++ b/emscripten.bzl
@@ -18,6 +18,7 @@ def xnnpack_emscripten_test_linkopts():
         "-s DEMANGLE_SUPPORT=1",
         "-s EXIT_RUNTIME=1",
         "-s ALLOW_MEMORY_GROWTH=1",
+        "-s USE_PTHREADS=0",
         "--pre-js $(location :preamble.js.lds)",
     ]
 
@@ -30,6 +31,7 @@ def xnnpack_emscripten_benchmark_linkopts():
         "-s EXIT_RUNTIME=1",
         "-s ALLOW_MEMORY_GROWTH=1",
         "-s TOTAL_MEMORY=445644800",  # 425M
+        "-s USE_PTHREADS=0",
         "--pre-js $(location :preamble.js.lds)",
     ]
 
